tasks/connectors/checkmarx_sast/checkmarx_sast.rb

```diff
@@ -0,0 +1,120 @@
+            { name: "checkmarx_sast_console_port",
+              type: "integer",
+              required: false,
+              default: nil,
+              description: "Your checkmarx_sast Console port, e.g. 8080" },
+            { name: "checkmarx_sast_user",
+              type: "user",
+              required: true,
+              default: nil,
+              description: "checkmarx_sast Username" },
+            { name: "checkmarx_sast_password",
+              type: "password",
+              required: true,
+              default: nil,
+              description: "checkmarx_sast Password" },
+            { name: "client_id",
+              type: "client detail",
+              required: true,
+              default: nil,
+              description: "client id of checkmarx SAST" },
+            { name: "client_secret",
+              type: "client secret ",
+              required: true,
+              default: nil,
+              description: "client secret of checkmarx SAST" },
+            { name: "grant_type",
+              type: "grant access type",
+              required: false,
+              default: "password",
+              description: "grant access type" },
+            { name: "scope",
+              type: "api scope",
+              required: false,
+              default: "access_control_api sast_api",
+              description: "scope API" },
+            { name: "kenna_api_key",
+              type: "api_key",
+              required: false,
+              default: nil,
+              description: "Kenna API Key" },
+            { name: "kenna_api_host",
+              type: "hostname",
+              required: false,
+              default: "api.kennasecurity.com",
+              description: "Kenna API Hostname" },
+            { name: "kenna_connector_id",
+              type: "integer",
+              required: false,
+              default: nil,
+              description: "If set, we'll try to upload to this connector" },
+            { name: "output_directory",
+              type: "filename",
+              required: false,
+              default: "output/checkmarx_sast",
+              description: "If set, will write a file upon completion. Path is relative to #{$basedir}" }
+          ]
+        }
+      end
+
+      def run(opts)
+        super # opts -> @options
+
+        initialze_options
+
+        # Request checkmarx sast auth api to get access token
+        token = request_checkmarx_sast_token
+        fail_task "Unable to authenticate with checkmarx_sast, please check credentials" unless token
+
+        # Request checkmarx sast api to fetch projects using token
+        projects = fetch_checkmarx_sast_projects(token)
+
+        projects.each do |project|
+          print_good "Project Name: #{project['name']}"
+          project_id = project["id"]
+          # Request checkmarx sast api to fetch all scans of each project
+          fetch_all_scans_of_project(token, project_id)
```

Comment:
@kdavesecurview @fshaiksecurview  I don't want to check this in until there is actually processing of the reports but this is a good start.